src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -20,11 +20,11 @@ namespace Stratis.Bitcoin.Consensus
     /// This component is an extension of <see cref="ConsensusManager"/> and is strongly linked to its functionality, it should never be called outside of CM.
     /// <para>
     /// View of the chains that are presented by connected peers might be incomplete because we always
-    /// receive only chunk of headers claimed by the peer in one message.
+    /// receive only a chunk of headers claimed by the peer in each message.
     /// </para>
     /// <para>
-    /// It is a role of the consensus manager to decide which of the presented chains is going to be treated as our best chain.
-    /// <see cref="ChainedHeaderTree"/> only advices which chains it might be interesting to download.
+    /// It is a role of the <see cref="ConsensusManager"/> to decide which of the presented chains is going to be treated as our best chain.
```

Comment:
It is the* role

Reply:
of maybe not actually

Reply:
If it's one of its roles, then "One of the roles of the <see cref="ConsensusManager"/> is to decide" would be clearer.

Reply:
Haha, I only changed the words consensus manager to the cref but I think that it has the same meaning without saying “one of...” just because it says it is the role of, doesn’t mean that is its only role.


A wider point on this documentation is that the interface is documented as if it were the concrete class and the implementation of the concrete class very explicitly mentioned on the interface.